src/blog/jwt-identification-and-hot-streams/index.md

```diff
@@ -0,0 +1,173 @@
+```
+
+Finally, we need to verify the token and extract the identifiers in the `#connect` method:
+
+```ruby
+module ApplicationCable
+  class Connection < ActionCable::Connection::Base
+    identified_by :current_user
+
+    def connect
+      token = request.params[:token]
+
+      decoded_token = JWT.decode token, JWT_ENCRYPTION_KEY, true, { algorithm: "HS256" }
+      # The #deserialize_identifiers method is a counterpart to #serialize_identifiers
+      # and is responsible for converting strings back to objects
+      identifiers = deserialize_identifiers(JSON.parse(decoded_token["ext"]))
+
+      # Populate identifiers
+      identifiers.each do |k, v|
+        self.public_send("#{k}=", v)
+      end
+    rescue JWT::DecodeError
+      reject_unauthorized_connection
+    end
+  end
+end
+```
+
+This doesn't look like such a simple solution, after all 😕 If you continue staring at this piece of code for a bit longer, you might notice that this implementation contains nothing specific to a particular application: we retrieve a token, verify it, and extract the identifiers. If this is the case, why can't we just re-implement this in Go? Well, we can!
+
+This is how AnyCable Go PRO JWT identification works:
+
+- A client passes a token along with the connection request (via a query string or a header).
+- A Go server retrieves a header, verifies its signature and TTL, and rejects the connection if token is invalid.
+- If the token is valid, we extract the identifiers, store them in the connection metadata and finalize the handshake (by sending a `{"type":"welcome"}` message).
+
+And we do not perform any RPC calls! Let's see how it affects the connection time:
+
+<figure class="blog--figure">
+<video loop="loop" muted="muted" autoplay="autoplay" playsinline="true" width="1221" height="800" class="blog--media">
+  <source src="./connect-benchmark.h264.mp4" ttype="video/mp4; codecs=avc1.4D401E,mp4a.40.2">
+</video>
+</figure>
+
+It's more than 2x faster now! And even more importantly, we've reduced the stress on the RPC server. This could be especially useful during _connection avalanches_.
+
+> JWT identification allows you to standardize the authentication flow for WebSockets, protects from cross-site WebSocket hijacking and results in a performance boost!
+
+Luckily, you don't need to write any of the code above yourself, even the token generation part a little companion gem, [anycable-rails-jwt][], which does this for you!
+
+The only question left is how to handle tokens expiration gracefully? AnyCable Go uses a specific disconnect reason to distinguish expiration from unauthorized access (sends a `{"type":"disconnect","reason":"token_expired"}` message). You can use it to refresh the token. Again, we're glad to provide an out-of-the-box solution for that in our brand new [anycable-client-refresh-tokens][]!
```

Comment:
**IMO, looks quite odd to ask a question as it's currently being done, my suggestion:**
The only question left is how to handle tokens expiration gracefully? --> "The only question left: how can we gracefully handle token expiration?"
**Next suggestion:**
AnyCable Go users a. specific disconnect reason --> AnyCable Go uses a specific disconnect message
**Third suggestion:**
solution for that --> solution for this